File: app/Models/Portfolio.php
<?php

declare(strict_types=1);

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Support\Str;

class Portfolio extends Model
{
    use HasFactory;

    protected $fillable = [
        'slug',
        'title',
        'excerpt',
        'description',
        'client_name',
        'service_id',
        'thumbnail',
        'gallery_images',
        'tech_stack',
        'challenge',
        'solution',
        'result',
        'project_url',
        'completed_at',
        'is_featured',
        'is_published',
        'sort_order',
    ];

    protected function casts(): array
    {
        return [
            'gallery_images' => 'array',
            'tech_stack' => 'array',
            'completed_at' => 'date',
            'is_featured' => 'boolean',
            'is_published' => 'boolean',
            'sort_order' => 'integer',
        ];
    }

    /**
     * Boot: auto-generate slug dari title.
     */
    protected static function boot(): void
    {
        parent::boot();

        static::creating(function (Portfolio $portfolio): void {
            if (empty($portfolio->slug)) {
                $portfolio->slug = Str::slug($portfolio->title);
            }
        });
    }

    /**
     * Get the service that owns the portfolio.
     */
    public function service(): BelongsTo
    {
        return $this->belongsTo(Service::class);
    }

    /**
     * Scope: only published portfolios.
     */
    public function scopePublished($query)
    {
        return $query->where('is_published', true);
    }

    /**
     * Scope: featured portfolios.
     */
    public function scopeFeatured($query)
    {
        return $query->where('is_featured', true);
    }

    /**
     * Get the thumbnail URL.
     */
    public function getThumbnailUrlAttribute(): ?string
    {
        return $this->thumbnail
            ? asset('storage/' . $this->thumbnail)
            : null;
    }
}

File: logikraf-v2/internal/domain/model/portfolio.go
package model

import "time"

type Portfolio struct {
	ID           uint      `gorm:"primaryKey" json:"id"`
	ServiceID    uint      `json:"service_id"`
	Title        string    `gorm:"size:255;not null" json:"title"`
	Slug         string    `gorm:"size:255;uniqueIndex;not null" json:"slug"`
	Excerpt      string    `gorm:"size:500" json:"excerpt"`
	Description  string    `gorm:"type:text" json:"description"`
	ClientName   string    `gorm:"size:255" json:"client_name"`
	ProjectURL   string    `gorm:"size:500" json:"project_url"`
	Thumbnail    string    `gorm:"size:500" json:"thumbnail"`
	IsPublished  bool      `gorm:"default:true" json:"is_published"`
	IsFeatured   bool      `gorm:"default:false" json:"is_featured"`
	SortOrder    int       `gorm:"default:0" json:"sort_order"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
	DeletedAt    *time.Time `gorm:"index" json:"-"`
}
